ext/pcre/php_pcre.c

```diff
@@ -678,11 +682,11 @@ PHPAPI pcre_cache_entry* pcre_get_compiled_regex_cache_ex(zend_string *regex, in
 			zend_string_release_ex(key, 0);
 		}
 		if (pp < ZSTR_VAL(regex) + ZSTR_LEN(regex)) {
-			php_error_docref(NULL,E_WARNING, "Null byte in regex");
+			zend_type_error("%s(): Regular expression cannot contain any null-bytes", get_active_function_name());
 		} else if (start_delimiter == end_delimiter) {
-			php_error_docref(NULL,E_WARNING, "No ending delimiter '%c' found", delimiter);
+			zend_value_error("%s(): Regular expression doesn't contain an ending delimiter \"%c\"", get_active_function_name(), delimiter);
 		} else {
-			php_error_docref(NULL,E_WARNING, "No ending matching delimiter '%c' found", delimiter);
+			zend_value_error("%s(): No ending matching delimiter \"%c\" found", get_active_function_name(), delimiter);
```

Comment:
I don't know what this message is exactly about

Reply:
From [the docs](https://www.php.net/manual/en/regexp.reference.delimiters.php):
> It is also possible to use bracket style delimiters where the opening and closing brackets are the starting and ending delimiter, respectively. `()`, `{}`, `[]` and `<>` are all valid bracket style delimiter pairs.
>```
>(this [is] a (pattern))
>{this [is] a (pattern)}
>[this [is] a (pattern)]
><this [is] a (pattern)>
>```
> Bracket style delimiters do not need to be escaped when they are used as meta characters within the pattern, but as with other delimiters they must be escaped when they are used as literal characters.

The "No ending matching delimiter" error here is raised when the ending delimiter for these bracket style delimiters cannot be found.

Reply:
@salathe Thank you very much! I was unaware of this feature, so I was surprised to see in a test that `<>` works. Unfortunately I couldn't connect this to the error message in question :)